include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -159,6 +310,72 @@ struct __parallel_sort_submitter<_IdType, __internal::__optional_kernel_name<_Le
     }
 };
 
+template <typename _Range, typename _Compare>
+struct __leaf_sorter_selector
+{
+    // 8 is the maximum reasonable value for bubble sub-group sorter due to algorithm complexity
+    // TODO: reconsider the value if another algorithm is used,
+    //       or an internal cap is set (e.g. sorting 2 sequences of 4/8 items each)
+    using _Leaf8 = __leaf_sorter<8, _Range, _Compare>;
+    using _Leaf4 = __leaf_sorter<4, _Range, _Compare>;
+    // 2 is the smallest reasonable value for merge-path group sorter since it loads 2 values at least
+    using _Leaf2 = __leaf_sorter<2, _Range, _Compare>;
+
+    using _Tp = oneapi::dpl::__internal::__value_t<_Range>;
+
+    std::variant<_Leaf8, _Leaf4, _Leaf2>
```

Comment:
My assumption that `variant` / `visit` pattern probably not required here due you return only one object from this function. Also, in the call of `std::visit` you have only one object and index.

Reply:
`variant`/`visit` has been removed in [0daae6e](https://github.com/oneapi-src/oneDPL/pull/1735/commits/0daae6e5e218f458c47d11c89600900673ea70de).

Reply:
Is these changes from the PR https://github.com/oneapi-src/oneDPL/pull/1735 ?